File: smartbuilding-web/api/permission_codes.py
"""
Aligné sur SmartBuilding.Shared.Constants.PermissionCodes (desktop).
"""

from __future__ import annotations

ROLE_PERMISSIONS: dict[str, list[str]] = {
    "Administrateur": ["*"],
    "PDG": ["*"],
    "Comptable": [
        "dashboard.view",
        "finance.manage",
        "finance.view",
        "location.manage",
        "suppliers.manage",
        "reports.export",
        "personnel.view",
    ],
    "Technique": [
        "dashboard.view",
        "technical.manage",
        "incidents.manage",
        "consumption.manage",
        "inventory.manage",
        "personnel.view",
    ],
    "Gestionnaire": [
        "dashboard.view",
        "location.manage",
        "visitors.manage",
        "incidents.manage",
        "personnel.manage",
        "consumption.manage",
        "email.manage",
        "reports.export",
    ],
}

ALL_PERMISSION_CODES = [
    "dashboard.view",
    "personnel.manage",
    "personnel.view",
    "technical.manage",
    "location.manage",
    "finance.manage",
    "finance.view",
    "suppliers.manage",
    "incidents.manage",
    "consumption.manage",
    "visitors.manage",
    "inventory.manage",
    "email.manage",
    "users.manage",
    "sync.manage",
    "reports.export",
]


def permissions_for_role(role: str) -> list[str]:
    perms = ROLE_PERMISSIONS.get(role, [])
    if "*" in perms:
        return list(ALL_PERMISSION_CODES)
    return list(perms)


def role_has_permission(role: str, code: str | None) -> bool:
    if not code:
        return True
    perms = ROLE_PERMISSIONS.get(role, [])
    if "*" in perms:
        return True
    return code in perms

File: SmartBuilding.Shared/Constants/PermissionCodes.cs
namespace SmartBuilding.Shared.Constants;

public static class PermissionCodes
{
    public const string DashboardView = "dashboard.view";
    public const string PersonnelManage = "personnel.manage";
    public const string PersonnelView = "personnel.view";
    public const string TechnicalManage = "technical.manage";
    public const string LocationManage = "location.manage";
    public const string FinanceManage = "finance.manage";
    public const string FinanceView = "finance.view";
    public const string SuppliersManage = "suppliers.manage";
    public const string IncidentsManage = "incidents.manage";
    public const string ConsumptionManage = "consumption.manage";
    public const string VisitorsManage = "visitors.manage";
    public const string InventoryManage = "inventory.manage";
    public const string EmailManage = "email.manage";
    public const string UsersManage = "users.manage";
    public const string SyncManage = "sync.manage";
    public const string ReportsExport = "reports.export";

    public static readonly IReadOnlyDictionary<string, string[]> RolePermissions = new Dictionary<string, string[]>
    {
        ["Administrateur"] = ["*"],
        ["Comptable"] =
        [
            DashboardView, FinanceManage, FinanceView, LocationManage,
            SuppliersManage, ReportsExport, PersonnelView
        ],
        ["Technique"] =
        [
            DashboardView, TechnicalManage, IncidentsManage,
            ConsumptionManage, InventoryManage, PersonnelView
        ],
        ["Gestionnaire"] =
        [
            DashboardView, LocationManage, VisitorsManage, IncidentsManage,
            PersonnelManage, ConsumptionManage, EmailManage, ReportsExport
        ],
        ["Réceptionniste"] = [VisitorsManage],
        ["Receptionniste"] = [VisitorsManage]
    };
}
